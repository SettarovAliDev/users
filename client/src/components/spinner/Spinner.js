import React from 'react';

import { SpinnerContainer, SpinnerOverlay } from './SpinnerStyles';

const Spinner = ({ size, big }) => {
  return (
    <SpinnerOverlay data-testid="spinner" big={big}>
      <SpinnerContainer size={size} />
    </SpinnerOverlay>
  );
};

export default Spinner;
