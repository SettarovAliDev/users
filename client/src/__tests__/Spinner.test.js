import React from 'react';
import renderer from 'react-test-renderer';
import { cleanup } from '@testing-library/react';
import 'jest-styled-components';

import Spinner from '../components/spinner/Spinner';

afterEach(cleanup);

it('renders spinner', () => {
  const tree = renderer.create(<Spinner size="7rem" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders big spinner', () => {
  const tree = renderer.create(<Spinner size="7rem" big />).toJSON();
  expect(tree).toMatchSnapshot();
});
