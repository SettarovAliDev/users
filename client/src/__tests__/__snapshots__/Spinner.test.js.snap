// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders big spinner 1`] = `
.c0 {
  height: 100vh;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  display: inline-block;
  width: 7rem;
  height: 7rem;
  border: 3px solid rgba(195,195,195,0.6);
  border-radius: 50%;
  border-top-color: #636767;
  -webkit-animation: fkRsEG 1s ease-in-out infinite;
  animation: fkRsEG 1s ease-in-out infinite;
  -webkit-animation: fkRsEG 1s ease-in-out infinite;
}

<div
  className="c0"
  data-testid="spinner"
>
  <div
    className="c1"
    size="7rem"
  />
</div>
`;

exports[`renders spinner 1`] = `
.c0 {
  height: 100%;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  display: inline-block;
  width: 7rem;
  height: 7rem;
  border: 3px solid rgba(195,195,195,0.6);
  border-radius: 50%;
  border-top-color: #636767;
  -webkit-animation: fkRsEG 1s ease-in-out infinite;
  animation: fkRsEG 1s ease-in-out infinite;
  -webkit-animation: fkRsEG 1s ease-in-out infinite;
}

<div
  className="c0"
  data-testid="spinner"
>
  <div
    className="c1"
    size="7rem"
  />
</div>
`;
